platform/engineering/design-docs/separate-content-build.md

```diff
@@ -0,0 +1,373 @@
+for more details.
+
+#### Pain points
+- Engineers encounter **confusing build errors** that are due to the mismatch in
+  outdated locally cached content and updated Liquid templates
+- Fetching content from the CMS and building HTML files when developing
+  applications **slows down the local builds** for engineers unnecessarily
+- When their applications have nothing to with the CMS
+- It **requires either SOCKS proxy access** or the `npm run fetch-drupal-cache`
+  command, which isn't obvious
+- **Application deployments and PR builds fail** when content in the CMS is
+  invalid
+  - This includes broken links and accessibility violations
+  - We have a safeguard that mostly works; if we run into a failure using the
+    latest CMS content, we try again with the cached content in S3 from the last
+    successful build
+    - This doesn't work when the GraphQL query has been updated; that
+      invalidates the cache
+    - Errors in liquid templates would potentially break the build and falling
+      back to the cache won't help
+
+### High Level Design
+The current front end build will be split up into two distinct builds:
+1. Content build
+    - Input: CMS content
+    - Output: HTML files, assets from the CMS (PDFs, images and the like)
+1. Application build
+    - Input: `vets-website` code
+    - Output: JavaScript and CSS bundles
+
+The output of both these builds will be deployed to separate buckets. The
+reverse proxies will route traffic intelligently to the appropriate bucket given
+the request.
+
+![Proposed traffic flow to S3 buckets](images/proposed-traffic-flow-to-s3-buckets.png)
+
+**Important note:** JavaScript application landing pages will live in the CMS.
+To get an application into production, the page in Drupal will need to be
+published.
+
+**Another important note:** There is no way to coordinate these two deploys to
+make an application live for the first time. The process will be to manually:
+1. Verify the application assets are live in production
+1. Publish the landing page
+1. Deploy a new content build
+
+## Specifics
+
+### Detailed Design
+
+#### Build and deploy
+- The content build will live in a separate repository
+  - **Question:** What should we call this?
+  - **Question:** Can the application build be run in Circle?
+  - **Question:** Where can we run the content build?
+- Content will be deployed to a separate S3 bucket from the application code
+- The S3 buckets will have versioned directories in them with the contents of
+  each deployment
+  - Old directories will be removed after 90 days
+    - **Question:** What makes sense? This is a totally arbitrary number
+
+#### Routing
+- The nginx will route their traffic based on some configuration file
+  - **Question:** What configuration file? How does it work?
+
+#### Testing changes to applications
+- The `webpack-dev-server` will create application landing pages for local
```

Comment:
this seems like something that needs a lot more details

Reply:
Short story: Currently, these landing pages are generated in the Metalsmith build. When we move the Metalsmith build out of `vets-websites`, it'll introduce a couple needs:
- Engineers still need those pages to serve up the applications they're building
- E2E tests still need to be run against "live" pages, not just pointed at JS bundles

To solve these problems, my idea was to make the dev server spin up temporary landing pages. These temporary pages would not be in production, but are meant to mirror it.

:thinking: There are probably other concerns this new process would introduce, but I'm not sure what all they are yet.

Is this the kind of detail you think should be added here, or are you thinking about something else entirely?

Reply:
> There are probably other concerns this new process would introduce, but I'm not sure what all they are yet.

That's my concern --- and that's what this design doc should address! So yes.

Go back to "you should be able to hand this doc off to a junior engineer and expect them to be able to implement it." Major questions / concerns should be cleared out already, and ideally there wouldn't be any unanticipated roadblocks left.